docs/architecture/ADR-003-Non-interactive-defaults.md

```diff
@@ -0,0 +1,691 @@
+# ADR 003: Non-interactive Defaults, Wrapped Transactions, and Subtree Root Message Inclusion Checks
+
+NOTE: Unlike normal tendermint/cosmos ADRs, this ADR isn't for deciding on whether or not we will implement non-interactive defaults. The goal of this document is to help reviewers and future readers understand what non-interactive defaults are, the considerations that went into the initial implementation, and how it differs from the original specs.
+
+The exact approach taken by the initial implementation, however, is certainly up for scrutiny.
+
+## Intro
+
+pls see the [original specs](https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md), from which this ADR paraphrases heavily.
+
+Currently, when checking for message inclusion, validators recreate the share commitment from the messages in the block and compare those with what are signed over in the `MsgPayForData` transactions also in that block. If any commitment is not found in one of the PFD transactions, or if there is a commitment that doesn't have a corresponding message, then they reject that block.
+
+While this functions as a message inclusion check, the light client has to assume that 2/3's of the voting power is honest in order to be assured that both the messages they are interested in and the rest of the messages paid for in that block are actually included. In order to have this property, we need a block validity rule where:
+
+> **All share commitments included in `MsgPayForData` must consist only of subtree roots of the data square.**
+
+The main issue with that requirement is that users must know the relevant subtree roots before they sign, which is problematic considering that if the block is not organized perfectly, the subtree roots will include data unknown to the user at the time of signing.
```

Comment:
[Non Blocking][Question]
which subtree roots? The root of the message shares?

Reply:
> which is problematic considering that if the block is not organized perfectly, the subtree roots will include data unknown to the user at the time of signing.

this is for messages that don't use the non-interactive defaults, there's no way to know which subtree roots to sign over ahead of time. The signers of the PFD would essentially have to know the arrangement of the square ahead of time.

Reply:
this comment, along with the other comment https://github.com/celestiaorg/celestia-app/pull/673#discussion_r962160480 confuses me, as they nullify eachother lol

we can sync on this super super quick if this still doesn't make sense. Perhaps I can fix some wording somewhere

Reply:
My bad, there are some nuances that I am still not grasping when it comes to generating the state roots. I will resolve this conversations and give them some more time then come back at you with questions if there are any.
Thanks a lot for your time, this ADR was extremely helpful, great work :rocket:

Reply:
thanks for all the great feedback! and sure thing. it took me like four times of learning and forgetting it before I got it lol

happy to answer any questions in our 1:1 or something similar.